Pa3/Durchführung/Project/app/State/ReadyToDrive.c

```diff
@@ -28,6 +28,25 @@
 
 EventEnum ReadyToDrive_CheckStateOfButtons(void)
 {
-
+    EventEnum retEvent = NO_EVENT_HAS_HAPPEND;
+
+    if (BUTTON_STATE_RELEASED == Button_getState(BUTTON_ID_A))
+    {
+        retEvent = START_BUTTON_HAS_BEEN_RELEASED;   
+    }
```

Comment:
See comment on enum
/** Logical button states (after debounce). */
typedef enum tag_ButtonStatus
this means it is a fixed state, not an action.
It will be in state Pressed as long as the button is physically pressed and in state released whenever the button is not pressed.
What you want to check is the action "released" so the state transition from Pressed to Released.
Therefore you have to check if the button was pressed first and then released.
```suggestion
    EventEnum retEvent = NO_EVENT_HAS_HAPPEND;
    
    static ButtonState lastStateButtonA = BUTTON_STATE_UNKNOWN; /* The static keyword makes it so the variable will keep it's last value even if the function is exited and called again */
    ButtonState activeState = Button_getState(BUTTON_ID_A);
    if((BUTTON_STATE_RELEASED == activeState) && (BUTTON_STATE_Pressed == lastStateButtonA))
    {
        retEvent = START_BUTTON_HAS_BEEN_RELEASED;
    }
    lastStateButtonA = activeState;
```

this could also be a static function with ButtonId as parameter instead of repeating it for every button